src/iov.c

```diff
@@ -63,24 +63,39 @@ uint64_t ofi_copy_iov_buf(const struct iovec *iov, size_t iov_count, uint64_t io
 void ofi_consume_iov_desc(struct iovec *iov, void **desc,
 			  size_t *iov_count, size_t to_consume)
 {
-	size_t i;
+	struct iovec *cur_iov;
+	void **cur_desc;
+
+	assert(to_consume && *iov_count);
+	assert(to_consume <= ofi_total_iov_len(iov, *iov_count));
+
+	if (*iov_count == 1 && iov[0].iov_len > to_consume) {
+		iov[0].iov_len -= to_consume;
+		return;
+	} else if (*iov_count == 1) {
+		*iov_count = 0;
+		return;
+	}
```

Comment:
I'd adjust the above to:
```
if (*iov_count == 1) {
    if (iov[0].iov_len == to_consume) {
        *iov_count = 0;
        return;
    }
    goto out;
}
```
See comment below.

Reply:
An if statement with an && condition technically ends up as multiple branches when converted to assembly.  My intent with the above was to avoid duplicating the same check twice (i.e. iov_count == 1)

Reply:
Ah, right, because of guarantee for short-circuit evaluation.  (Not true for fortran!)